src/Player/PlayerProxy.cpp

```diff
@@ -48,6 +52,48 @@ void PlayerProxy::setOnClose(const function<void(const SockException &ex)> &cb)
     _on_close = cb ? cb : [](const SockException &) {};
 }
 
+void PlayerProxy::setOnDisconnect(const std::function<void()> &cb)
+{
+    _on_disconnect = cb ? cb : []() {};
+}
+
+void PlayerProxy::setOnConnect(const std::function<void(const TranslationInfo&)> &cb)
+{
+    _on_connect = cb ? cb : [](const TranslationInfo&) {};
+}
+
+void PlayerProxy::setTranslationInfo()
+{
+    _transtalion_info.byte_speed = _media_src->getBytesSpeed();
+    _transtalion_info.start_time_stamp = _media_src->getCreateStamp();
+    _transtalion_info.stream_info.clear();
+    auto tracks = _muxer->getTracks();
+    for (auto &track : tracks) {
+        _transtalion_info.stream_info.emplace_back();
+        _transtalion_info.stream_info[_transtalion_info.stream_info.size() - 1].bitrate = track->getBitRate();
```

Comment:
How about make a `int index` to save `_transtalion_info.stream_info.size() - 1` ?

Reply:
I can do it but I think this method is handy
Of course, performance will be higher if done through int index
I will do it soon

Reply:
How about this 
```c++
_transtalion_info.stream_info.emplace_back();
auto &back = _transtalion_info.stream_info.back();
back.bitrate = ....
```